FC-Chess/FC-Chess.cpp

```diff
@@ -107,28 +105,271 @@ void FCC::Chess::checkMovesHorizontal()
+    i -= 1, j += 3;
+    if (checkIsPosInBounds(&i, &j)) 
+    {
+        if (!checkIsFriendlyOnField(&i, &j)) posPossible[i][j] = true;
+    }
+
+    i = selX, j = selY;     // resetting variables
+    i -= 1, j -= 3;
+    if (checkIsPosInBounds(&i, &j)) 
+    {
+        if (!checkIsFriendlyOnField(&i, &j)) posPossible[i][j] = true;
+    }
+
+    i = selX, j = selY;     // resetting variables
+    i += 3, j += 1;
+    if (checkIsPosInBounds(&i, &j)) 
+    {
+        if (!checkIsFriendlyOnField(&i, &j)) posPossible[i][j] = true;
+    }
+
+    i = selX, j = selY;     // resetting variables
+    i += 3, j -= 1;
+    if (checkIsPosInBounds(&i, &j)) 
+    {
+        if (!checkIsFriendlyOnField(&i, &j)) posPossible[i][j] = true;
+    }
+
+    i = selX, j = selY;     // resetting variables
+    i -= 3, j += 1;
+    if (checkIsPosInBounds(&i, &j)) 
+    {
+        if (!checkIsFriendlyOnField(&i, &j)) posPossible[i][j] = true;
+    }
+
+    i = selX, j = selY;     // resetting variables
+    i -= 3, j -= 1;
+    if (checkIsPosInBounds(&i, &j)) 
+    {
+        if (!checkIsFriendlyOnField(&i, &j)) posPossible[i][j] = true;
+    }
+}
+
+void FCC::Chess::checkMovesKing()
+{
+    // creating variables for x and y
+    uint8_t i, j;
+
+    // checking possible moves
+    i = selX - 1, j = selY + 1;
+    for (; i <= selX + 1; i++)
+    {
+        if (checkIsPosInBounds(&i, &j))
+        {
+            if (!checkIsFriendlyOnField(&i, &j)) posPossible[i][j] = true;
+        }
     }
+
+    i = selX - 1, j = selY - 1;
+    for (; i <= selX + 1; i++)
+    {
+        if (checkIsPosInBounds(&i, &j))
+        {
+            if (!checkIsFriendlyOnField(&i, &j)) posPossible[i][j] = true;
+        }
+    }
+
+    i = selX - 1, j = selY;
+    if (checkIsPosInBounds(&i, &j))
+    {
+        if (!checkIsFriendlyOnField(&i, &j)) posPossible[i][j] = true;
+    }
+
+    i = selX + 1, j = selY;
+    if (checkIsPosInBounds(&i, &j))
+    {
+        if (!checkIsFriendlyOnField(&i, &j)) posPossible[i][j] = true;
+    }
+}
+
+void FCC::Chess::checkMovesPawn()
+{
+    // black pieces
+    if (posColor[selX][selY] == FCC::PiecesColors::BLACK)
+    {
+        // creating variables for x and y
+        uint8_t i, j;
+
+        // checking possible moves
+        i = selX - 1, j = selY - 1;
+        for (; i <= selX + 1;)
+        {
+            if (checkIsPosInBounds(&i, &j))
+            {
+                if (checkIsEnemyOnField(&i, &j)) posPossible[i][j] = true;
+            }
+            i += 2;
+        }
+
+        i = selX, j = selY - 1;
+        if (checkIsPosInBounds(&i, &j))
+        {
+            if (checkIsPositionClear(&i, &j)) posPossible[i][j] = true;
+        }
+    }
+
+    // white pieces
+    if (posColor[selX][selY] == FCC::PiecesColors::WHITE)
+    {
+        // creating variables for x and y
+        uint8_t i, j;
+
+        // checking possible moves
+        i = selX - 1, j = selY + 1;
+        for (; i <= selX + 1;)
```

Comment:
Use a 2D loop

Reply:
in the upcoming push, I removed the j variable. So that would mean one more variable. Does it make sense?

Snippet from upcoming push:
`void FCC::Chess::checkMovesPawn()
{
    // black pieces
    if (posColor[selX][selY] == FCC::PiecesColors::BLACK)
    {
        // checking possible moves
        // left and right for enemy pieces
        for (uint8_t i = selX - 1; i <= selX + 1; i += 2)
        {
            if (checkIsPosInBounds(i, selY - 1))
            {
                if (checkIsEnemyOnField(i, selY - 1)) posPossible[i][selY - 1] = true;
            }
        }

        // standard move to the front
        if (checkIsPosInBounds(selX, selY - 1))
        {
            if (checkIsPositionClear(selX, selY - 1)) posPossible[selX][selY - 1] = true;
        }
    }

    // white pieces
    if (posColor[selX][selY] == FCC::PiecesColors::WHITE)
    {
        // checking possible moves
        // left and right for enemy pieces
        for (uint8_t i = selX - 1; i <= selX + 1; i += 2)
        {
            if (checkIsPosInBounds(i, selY + 1))
            {
                if (checkIsEnemyOnField(i, selY + 1)) posPossible[i][selY + 1] = true;
            }
        }

        // standard move to the front
        if (checkIsPosInBounds(selX, selY + 1))
        {
            if (checkIsPositionClear(selX, selY + 1)) posPossible[selX][selY + 1] = true;
        }
    }
}`

Reply:
this should work ^^